virl2_client/models/auth_management.py

```diff
@@ -0,0 +1,391 @@
+    _CONFIG_URL = _BASE_URL + "/config"
+    _TEST_URL = _BASE_URL + "/test"
+
+    def __init__(self, auth_management: AuthManagement):
+        self._auth_management = auth_management
+        self._last_sync_time = 0.0
+        self._settings = {}
+
+    def sync_if_outdated(self) -> None:
+        timestamp = time.time()
+        if (
+            self._auth_management.auto_sync
+            and timestamp - self._last_sync_time
+            > self._auth_management.auto_sync_interval
+        ):
+            self.sync()
+
+    def sync(self) -> None:
+        self._settings = self._get_current_settings()
+        self._last_sync_time = time.time()
+
+    @property
+    def _session(self) -> Client:
+        return self._auth_management._session
+
+    def _get_current_settings(self) -> dict:
+        settings = self._session.get(self._CONFIG_URL).json()
+        del settings["method"]
+        return settings
+
+    def get_setting(self, setting: str) -> Any:
+        """
+        Returns the value of a specific setting of this authentication method.
+        Note: consider using parameters instead.
+        """
+        self.sync_if_outdated()
+        return self._settings[setting]
+
+    def get_settings(self) -> dict:
+        """Returns a dictionary of the settings of this authentication method."""
+        self.sync_if_outdated()
+        return self._settings.copy()
+
+    def _update_setting(self, setting: str, value) -> None:
+        """
+        Update a setting for this auth method.
+
+        :param setting: the setting to update
+        :param value: the value to set the setting to
+        """
+        if setting == "method":
+            raise ValueError(
+                f"To set method, use 'AuthManagement.method = {value}' instead."
+            )
+        # TODO: currently, method is required, so we have to provide it;
+        #  do we provide this one or the one that is currently active?
+        #  in other words, if method=local and you change an LDAP setting, do we
+        #  switch to LDAP or keep local?
+        #  we can also provide an argument, so the user can choose, but what would
+        #  the default be?
+        # also note this should generally not happen;
+        # see AuthManagement.method above
+        method = self._METHOD
+        self._session.put(
+            self._CONFIG_URL,
+            json={setting: value, "method": method},
+        )
+        self._auth_management._method = AuthMethod(method)
+        if setting in self._settings:
+            self._settings[setting] = value
+
+    def update_settings(self, settings_dict: Optional[dict] = None, **kwargs) -> None:
+        """
+        Update multiple auth settings at once.
+        More efficient in batches than setting manager properties.
+        If passed a dictionary, reads the dictionary for settings.
+        If passed multiple keyword arguments, each is taken as a setting.
+        Can mix and match; keywords take precedence.
```

Comment:
I think that this is an overkill for a method that should be used once at max if ever :) but sure

Reply:
It is not that unnecessary, I would still keep it. However I think we should be able to switch method with this one too, if we provide "method" as an argument.

Reply:
1. The method is definitely useful. Might be a bit overengineered, but you can e.g. pass it a dictionary of settings but override one using a keyword.
2. Strong disagree on allowing 'method' as argument. This is a method of managers. If you use LDAPManager, you should not be able to use it to switch to 'local'; that is what we already agreed on, isn't is?